.github/workflows/publish-packages.yml

```
@@ -0,0 +1,106 @@
+name: Publish packages
+
+on:
+  workflow_dispatch:
+    inputs:
+      tag:
+        type: string
+        required: true
+        description: |
+          github release tag, branch specific examples:
+            - `main`: v1.0.1 -> can only be run from `main`
+            - `beta`: v1.0.2-beta0 -> `beta` can only be run from `beta`
+            - `*`: v1.0.2-test0 -> use format v1.0.2-[NPM_TAG][NUMBER]"
+
+jobs:
+  publish:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Install moreutils
+        run: sudo apt-get install -y moreutils
+
+      - name: Install git-toolbelt
+        run: |
+          wget https://github.com/nvie/git-toolbelt/archive/v1.8.0.tar.gz
+          tar -xzf v1.8.0.tar.gz
+          mkdir -p "$HOME/.local/bin"
+          cp git-toolbelt-1.8.0/git-* "$HOME/.local/bin"
+          rm -rf v1.8.0.tar.gz git-toolbelt-1.8.0
+          echo "$HOME/.local/bin" >> $GITHUB_PATH
+
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Validate git tag
+        run: ./.github/scripts/validate-version.sh "${{ inputs.tag }}"
+
+      - name: Set release version
+        id: version
+        env:
+          GIT_TAG: ${{ inputs.tag }}
+        run: echo "value=${GIT_TAG#v}" >> $GITHUB_OUTPUT
+
+      - name: Set npm tag
+        id: npm_tag
+        env:
+          GIT_TAG: ${{ inputs.tag }}
+        run: |
+          get_npm_tag () {
+            if grep -q "-" <<< "$1"; then
+                echo "${1##*-}" | sed 's/[0-9]//g'
+            else
+                echo "latest"
+            fi
+          }
+          echo value=$(get_npm_tag "$GIT_TAG") >> $GITHUB_OUTPUT
+
+      - name: Check values
+        run: |
+          echo ${{ inputs.tag }}
+          echo ${{ steps.version.outputs.value }}
+          echo ${{ steps.npm_tag.outputs.value }}
```

Always quote shell arguments that contain variables:

```suggestion
          echo "${{ inputs.tag }}"
          echo "${{ steps.version.outputs.value }}"
          echo "${{ steps.npm_tag.outputs.value }}"
```